### PR TITLE
CTW-1445 Showing subscription info in the patient overview

### PR DESCRIPTION
**Pull request by eric-gunther**
Adding patient subscription information to the overview page.

The most notable, potentially contentious, thing here is how we're deriving the subscription description/bullets from the package metadata. Will bring in DA for an opinion.

Pasted Graphic

### CODE REVIEW

File: src/utils/request.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export type CTWRequestInit = {
 
 export function ctwFetch(input: RequestInfo | URL, init?: CTWRequestInit): Promise<Response> {
   const headers = init?.headers || {};
-  headers["Zus-Request-Source"] = "component-library";
+  // headers["Zus-Request-Source"] = "component-library";
```
Comment (luthfib) [inline src/utils/request.ts:29]: Im guessing you don't want to do this anymore?

File: src/components/content/patient-history/patient-history-action.tsx [hunk 2]
```diff
@@ -3,15 +3,22 @@ import { RequestRecordsButton } from "./request-records-button";
...
 export const PatientHistoryLastRetrievedWithAction = ({
   hideRequestRecords = false,
+  includePatientDemographicsForm = false,
 }: PatientHistoryActionProps) => (
   <>
     <div>
       <PatientHistoryLastRetrieved />
     </div>
-    <div>{!hideRequestRecords && <RequestRecordsButton />}</div>
+    {/* TODO - hide demographics! */}
```
Comment (luthfib) [inline src/components/content/patient-history/patient-history-action.tsx:17]: Leaving a comment here to ensure we come back to this

File: src/components/content/overview/patient-subscription.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        // @ts-ignore
+        i18next.exists(x) ? i18next.t(x) : x
+      ) ?? [],
+    recurringProvidersWithInterval:
+      patientSubscription.package?.meta.recurringProvidersWithInterval?.map((x) => ({
+        // @ts-ignore
+        provider: i18next.exists(x.provider) ? i18next.t(x.provider) : x.provider,
+        intervalDays: x.intervalDays,
+      })) ?? [],
+  };
```
Comment (luthfib) [inline src/components/content/overview/patient-subscription.tsx:97]: Curious why we had to have all these `ts-ignore`. Might have to pull down to understand the issue

Comment (eric-gunther) [reply src/components/content/overview/patient-subscription.tsx:76]: Yeah kinda ugly, the call to `i18next.t` causes a Typescript error even though it's within a conditional that checks to see if `x` exists in the translation dictionary.

Comment (eric-gunther) [reply src/components/content/overview/patient-subscription.tsx:77]: I added a comment explaining.

File: src/services/subscriptions/subscriptions.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    description: string;
+    meta: PackageMeta;
+  };
+};
+
+export type PackageMeta = {
+  freshmakerProviders?: string[];
+  initialProviders?: string[];
+  recurringProvidersWithInterval?: {
+    intervalDays: number;
```
Comment (eric-gunther) [inline src/services/subscriptions/subscriptions.ts:22]: What are the other interval values?

Comment (eric-gunther) [reply src/services/subscriptions/subscriptions.ts:22]: Need to address this before merging

File: src/components/content/overview/patient-subscription.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
+import i18next from "i18next";
+import { Loading } from "@/components/core/loading";
+import { ErrorAlert, usePatient } from "@/index";
+import {
+  PatientSubscription,
+  usePatientSubscription,
+} from "@/services/subscriptions/subscriptions";
+import { compact, uniq } from "@/utils/nodash";
+
+const demoPatientSubscription = {
```
Comment (eric-gunther) [inline src/components/content/overview/patient-subscription.tsx:10]: Is this too hack? Setting up a mock package so that it looks better for test/demo patients.

File: src/fhir/diagnostic-report.ts [hunk 6]
```diff
@@ -149,7 +149,7 @@ export async function fetchDiagnosticReportsFromFQSById(
     const graphClient = createGraphqlClient(requestContext);
     const { data } = await fqsRequest<DiagnosticReportGraphqlResponse>(
       graphClient,
-      getDiagnosticReportQuery(false),
+      getDiagnosticReportQuery(true),
```
Comment (eric-gunther) [inline src/fhir/diagnostic-report.ts:152]: Ended up moving this change to its own PR, small fix.

File: src/components/content/overview/patient-subscription.tsx (hunk 5 above)
Comment (luthfib) [reply src/components/content/overview/patient-subscription.tsx:10]: `isTestPatient` where does this property come from? Is this something the api returns? Generally don't like cases like this. From what I understand this data is particularly hard to mock

File: src/components/content/overview/patient-subscription.tsx [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+};
+
+// Take subscription metadata and return a list of translated data sources and the details on their acquisition models
+export const getDataSources = (patientSubscription: PatientSubscription) => {
+  const translatedPackages = {
+    freshmakerProviders:
+      patientSubscription.package?.meta.freshmakerProviders?.map((x) =>
+        // All of these ts-ignore's are because the call to i18next.t has a TS error even though there is a check that x exists in the dictionary
+        // @ts-ignore
+        i18next.exists(x) ? i18next.t(x) : x
```
Comment (luthfib) [inline src/components/content/overview/patient-subscription.tsx:81]: You could define at type called like 

```export type Providers = "commonwell" | "surescripts" | "bamboo" | "collective" | "quest"```


and do `i18next.exists(x) ? i18next.t(x as Providers) : x`

Essentially what is going on it it knows all the types in `resources.en.main` but sees that these ones are strings and thus not matching these types. It's still slightly messy, but at least we could catch other ts errors

File: src/components/content/overview/patient-subscription.tsx (hunk 5 above)
Comment (eric-gunther) [reply src/components/content/overview/patient-subscription.tsx:10]: It's a getter on the patient model checking for a test tag that's a pretty reliable indicator that the patient is for test/demo. The subscription and patient history data has definitely been challenging to mock and in pretty much all cases it's just empty and makes the demo somewhat confusing. It's certainly easier for us to mock the data close to the UI like this, and it doesn't need to be particularly intelligent (ie. it can always just show the same thing), but I wonder if this is introducing a bad pattern. I'll poke the team about it in Slack.

**Review by luthfib — APPROVED** [review]
LGTM.
